### PR TITLE
update kubectl-rook-ceph repo

### PR DESCRIPTION
**Pull request by yati1998**
This commit updates the kubectl-rook-ceph repo to include latest changes.

### CODE REVIEW

**Comment by yati1998** [comment]
@subhamkrai please review the PR

File: go.mod [hunk 1]
```diff
@@ -1,14 +1,14 @@
 module github.com/red-hat-storage/odf-cli
 
-go 1.21
+go 1.20
```
Comment (travisn) [inline go.mod:3]: Let's not downgrade the go version

**Review by travisn — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (yati1998) [reply go.mod:3]: oops, Sorry. I thought I reverted it.

File: go.mod [hunk 2]
```diff
@@ -4,11 +4,11 @@ go 1.21
 
 require (
 	github.com/pkg/errors v0.9.1
-	github.com/rook/kubectl-rook-ceph v0.6.1-0.20231211183203-70ede2704719
-	github.com/rook/rook v1.13.0
+	github.com/rook/kubectl-rook-ceph v0.7.1-0.20240129162458-c33feeeaf0fb
```
Comment (travisn) [inline go.mod:7]: This includes all the latest commits to the upstream repo, right?

Comment (yati1998) [reply go.mod:7]: yes

**Review by travisn — APPROVED** [review]
/approve
/lgtm

**Pull request merged**